packages/doppler-v3-sdk/src/entities/factory/ReadWriteFactory.ts

```
@@ -529,6 +590,22 @@ export class ReadWriteFactory extends ReadFactory {
       liquidityMigrator,
     } = contracts;
 
+    let minedSalt = this.generateRandomSalt(userAddress);
+    for (let salt = BigInt(0); salt < BigInt(1_000_000); salt++) {
+      const saltBytes = `0x${salt.toString(16).padStart(64, "0")}` as Hash;
+
+      const token = this.computeCreate2Address(
+        saltBytes,
+        tokenInitHash,
+        tokenFactory
+      );
+      if (token.toLowerCase().endsWith(VANITY_ADDRESS_ENDING)) {
+        console.log(token);
+        minedSalt = saltBytes;
+        break;
+      }
+    }
+
```

we should set a success boolean somewhere before the loop, set it to true if we find a valid salt, and then check it after the loop, and maybe throw an error if one isnt found, or just return a 0 val

otherwise if we dont find a valid salt in 1M generations this will just return the last salt it tried without telling the user anything

1M iterations is a lot so this may be extremely unlikely lol but it doesnt cost us much to add it